### PR TITLE
fix multiple authcontext directive

### LINKED ISSUE #42: feature: multiple directive
It would be great to support multiple directives:

- `@hasRole`
- `@hasPermission`

Example:

```
type Article {
  id: ID!
  title: String!
  content: String! @hasPermission(resources: ["subscribed"])
}

type Query {
  listArticles: [Article]!
}

type Mutation {
  publishArticle(articleId: ID!): Article! @hasRole(role: "editor")
  unpublishArticle(articleId: ID!):Boolean @hasPermission(resources: ["publish","delete"])
}
```

Doing so, it should be open to integrating external Identity Management such as:

- [keycloak](https://www.keycloak.org/)
- [Auth0 Authorization extension](https://auth0.com/docs/extensions/authorization-extension)
- any Identity Manager that implements the User -> Role -> Permission paradigm

### PR DESCRIPTION
**Pull request by Eomm**
closes #42 

It was already possible to register multiple directives but it was not documented (but it was tested)

I have added a new readme section and a test where I spotted a small bug when multiple authContext are defined.

### CODE REVIEW

File: test/advanced.js [hunk 1]
```diff
@@ -174,7 +174,7 @@ test('should be able to access the query to determine that users have insufficie
 })
 
 test('should support being registered multiple times', async (t) => {
```
Comment (jonnydgreen) [inline test/advanced.js:176]: could we have a separate test for the `authMerge` fix rather than changing an existing test?

**Comment by jonnydgreen** [comment]
Good find with the auth merge issue - just a couple of comments :)

**Review by mcollina — APPROVED** [review]
good work! LGTM

**Review by jonnydgreen — APPROVED** [review]
Seconded! LGTM

**Pull request merged**